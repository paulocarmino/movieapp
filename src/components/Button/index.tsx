import React from "react"

export type ButtonProps = {
  icon?: JSX.Element | null
} & React.ComponentPropsWithRef<'button'>

const Button = React.forwardRef<HTMLButtonElement, ButtonProps>((
  {
    icon,
    children,
    ...rest
  },
  ref
) => {
  return (
    <button
      ref={ref}
      type="button"
      className="inline-flex items-center py-2 px-3 text-sm leading-4 text-gray-400 hover:bg-gray-800 rounded-md border border-gray-400 focus:outline-none shadow-sm"
      {...rest}
    >
      {!!icon && (
        <div data-testid="icon-button" className="mr-2 -ml-0.5 max-h-5 max-w-5">
          {icon}
        </div>
      )}
      {children}
    </button>
  )
})

export default Button
