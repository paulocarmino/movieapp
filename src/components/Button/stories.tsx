import { Story, Meta } from '@storybook/react'
import { AiOutlineHeart } from 'react-icons/ai'

import Button from '.'

export default {
  title: 'Button',
  component: Button,
  argTypes: {
    icon: {},
    children: {
      type: 'string'
    }
  }
} as Meta

export const Default: Story = (args) => <Button {...args}></Button>

Default.args = {
  children: 'Pesquisar'
}


export const WithIcon: Story = (args) => <Button icon={<AiOutlineHeart />} {...args}></Button>

WithIcon.args = {
  children: 'Add aos favoritos'
}
