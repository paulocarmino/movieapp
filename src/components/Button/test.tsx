import { render, screen } from '@testing-library/react'
import { AiOutlineHeart } from 'react-icons/ai'

import Button from '.'

describe('<Button />', () => {
  it('should render the heading', () => {
    const { container } = render(<Button>Login</Button>)

    expect(screen.getByRole('button', { name: /Login/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('deve mostrar o icone quando passado como prop', () => {
    render(<Button icon={<AiOutlineHeart />}>Login</Button>)

    expect(screen.getByTestId('icon-button')).toBeInTheDocument()
  })
})
