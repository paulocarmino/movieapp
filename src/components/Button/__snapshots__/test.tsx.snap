// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render the heading 1`] = `
<button
  class="inline-flex items-center py-2 px-3 text-sm leading-4 text-gray-100 hover:bg-gray-800 rounded-md border border-gray-100 focus:outline-none shadow-sm"
  type="button"
>
  Login
</button>
`;
